pkg/client/client.go

```diff
@@ -353,12 +382,22 @@ func (c *client) Get(ctx context.Context, key ObjectKey, obj Object, opts ...Get
 	}
 }
 
+// shouldReturnCacheErr determines if we should return the error we got from the cache, which is always except
+// for the case that we're configured to do live lookups and got a cache.ErrResourceNotCached
```

Comment:
If we're doing live lookups, why would we get a `cache.ErrResourceNotCached`?

Reply:
Will reword, is not clear